staging/src/k8s.io/apimachinery/pkg/util/json/internal/third_party/golang/encoding/json/decode.go

```diff
@@ -0,0 +1,1345 @@
+// Copyright 2010 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Represents JSON data structure using native Go types: booleans, floats,
+// strings, arrays, and maps.
+
+package json
+
+import (
```

Comment:
Just noting we probably should add some tests for the new features/cases before merging.

Reply:
Yup, that's why the commit is still WIP